src/pages/QuizResults.tsx

```diff
@@ -2,24 +2,91 @@ import MyButton from '../components/UI/button/MyButton';
 import '../App.css';
 import { useNavigate } from 'react-router-dom';
 import { paths } from '../utils/paths';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { clearQuizState, setStartTime } from '../slices/quizSettingsSlice';
+import { clearResults } from '../slices/quizResultsSlice';
+
+const categoryMap: { [key: number]: string } = {
+  9: 'General Knowledge',
+  10: 'Books',
+  11: 'Film',
+  12: 'Music',
+  13: 'Musicals and Theatres',
+  14: 'Video Games',
+  15: 'Board Games',
+  16: 'Science & Nature',
+  17: 'Science: Computers',
+  18: 'Science: Gadgets',
+  19: 'Science: Electronics',
+  20: 'Science: Physics',
+  21: 'Science: Biology',
+  22: 'Politics',
+  23: 'Art',
+  24: 'Animals',
+  25: 'Vehicles',
+  26: 'History',
+  27: 'Celebrities',
+  28: 'Sports',
+  29: 'Mythology',
+  30: 'Geography',
+  31: 'Entertainment: Video Games',
+  32: 'Entertainment: Board Games',
+  33: 'Entertainment: Comics',
+  34: 'Entertainment: Japanese Anime & Manga',
+  35: 'Entertainment: Cartoon & Animations',
+};
 
 const QuizResults: React.FC = () => {
   const navigate = useNavigate();
 
+  const dispatch = useAppDispatch();
+
   const handleRestartButton = () => {
+    dispatch(clearResults());
+    dispatch(setStartTime(Date.now()));
     navigate(paths.main);
   };
 
   const handleAnotherQuizButton = () => {
+    dispatch(clearQuizState());
+    dispatch(clearResults());
     navigate(paths.home);
   };
 
+  const questions = useAppSelector((state) => state.quizSettings.questions);
+
+  const correctAnswersCount = useAppSelector(
+    (state) => state.quizResults.correctAnswers,
+  );
+
+  const configuration = useAppSelector(
+    (state) => state.quizSettings.configuration,
+  );
+
+  const startTime = useAppSelector((state) => state.quizSettings.startTime);
+
+  const endTime = useAppSelector((state) => state.quizSettings.endTime);
```

Comment:
```suggestion
  const { questions, correctAnswers: correctAnswersCount, configuration, startTime, endTime } = useAppSelector((state) => state.quizSettings.questions);

```


 it does the same but with one useAppSelector call